comm.cpp

```diff
@@ -320,17 +321,24 @@ int UdpEndpoint::open(const char *ip, unsigned long port)
     sockaddr.sin_addr.s_addr = inet_addr(ip);
     sockaddr.sin_port = htons(port);
 
-    if (setsockopt(fd, SOL_SOCKET, SO_BROADCAST, &broadcast_val, sizeof(broadcast_val))) {
-        log_error_errno(errno, "Error enabling broadcast in socket (%m)");
-        goto fail;
+    if (to_bind) {
```

Comment:
Well... without binding how would the OS know that this socket is here to receive UDP packets on any given port? Don't forget that it's a different process (simulator) that is sending the UDP packets.
Without binding it just doesn't work: mavlink_router just don't receive any UDP packet.